api.go

```diff
@@ -528,6 +532,11 @@ func (c Client) executeMethod(method string, metadata requestMetadata) (res *htt
 		// For all other cases break out of the retry loop.
 		break
 	}
+
+	if retryCount >= MaxRetry {
+		err = errors.New("Max retry reached")
```

Comment:
Yes that is what we need here, we should only return the last error after all the retries.